src/App.tsx

```diff
@@ -1,93 +1,136 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import { useState, useEffect } from 'react';
+import { HashRouter as Router } from 'react-router-dom';
+
+import { AddTodoForm } from './components/AddTodoForm';
+import { TodoList } from './components/TodoList';
+import { TodosFilter } from './components/TodosFilter';
+import { Todo } from './types/types';
 
 export const App: React.FC = () => {
+  const [inputText, setInputText] = useState('');
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [status, setStatus] = useState('all');
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>([]);
+
+  const filterHandler = () => {
+    switch (status) {
+      case 'completed':
+        setFilteredTodos(todos.filter(todo => todo.completed));
+        break;
+      case 'active':
+        setFilteredTodos(todos.filter(todo => !todo.completed));
+        break;
+      default:
+        setFilteredTodos(todos);
+    }
+  };
+
+  const saveLocalTodos = () => {
+    localStorage.setItem('todos', JSON.stringify(todos));
+  };
+
+  const getLocalTodos = () => {
+    if (localStorage.getItem('todos') === null) {
+      localStorage.setItem('todos', JSON.stringify([]));
+    } else {
+      setTodos(JSON.parse(localStorage.getItem('todos') || ''));
+    }
+  };
+
+  const countCompleted = () => {
+    let quantityCompleted = 0;
+
+    todos.forEach(todo => {
+      if (todo.completed) {
+        quantityCompleted += 1;
+      }
+    });
+
+    return todos.length - quantityCompleted;
+  };
+
+  const clearCompleted = () => {
+    setTodos(todos.filter(todo => !todo.completed));
+  };
+
+  const handlingUpdate = (todoId: number, newTitle: string) => {
+    setTodos(todos.map(todo => {
+      if (todoId !== todo.id) {
+        return todo;
+      }
+
+      return {
+        ...todo,
+        title: newTitle,
+      };
+    }));
+  };
+
+  const mainCount = todos.length - countCompleted();
+
+  const toggleAll = () => {
+    setTodos(todos.map(todo => {
+      if (mainCount < todos.length) {
+        if (!todo.completed) {
+          return {
+            ...todo,
+            completed: true,
+          };
+        }
+
+        return todo;
+      }
+
+      return {
+        ...todo,
+        completed: !todo.completed,
+      };
+    }));
```

Comment:
it's tooo complicated, you have a checkmark for that, so either set all completed/ not completed status based on a checkmark state